### PR TITLE
[FIX] website_profile: show + only when gained karma is positive

### PR DESCRIPTION
**Pull request by niyasraphy**
before this commit, in the website profile, ie in the forum, if we filter the users using This week or This month filters, always the + is displayed along with the gained karma.

so if the user's gained karma during the selected period is negative values, lets say -500, currently it displays as follows -->  +-500 , instead of -500

after this commit, if the gained karma during the period is it will show as -500 instead of +-500

Before:
Screenshot from 2023-02-01 20-57-16

After:
Screenshot from 2023-02-01 20-56-47

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_profile/views/website_profile.xml [hunk 1]
```diff
@@ -561,7 +561,7 @@
                 <span class="badge badge-danger font-weight-normal px-2" t-if="not user['website_published']">Unpublished</span>
                 <strong class="text-muted" t-esc="user['rank']"/>
                 <div class="h3 my-2" t-if="user['karma_gain']">
-                    <span class="badge badge-pill badge-success px-3 py-2" >+ <t t-esc="user['karma_gain']"/> XP</span>
+                    <span class="badge badge-pill badge-success px-3 py-2" ><t t-if="user['karma_gain'] > 0">+ </t><t t-esc="user['karma_gain']"/> XP</span>
```
Comment (MiquelRForgeFlow) [inline addons/website_profile/views/website_profile.xml:564]:
```suggestion
                    <span class="badge badge-pill badge-success px-3 py-2" t-if="user['karma_gain'] > 0">+ <t t-esc="user['karma_gain']"/> XP</span>
                    <span class="badge badge-pill badge-warning px-3 py-2" t-if="user['karma_gain'] < 0"><t t-esc="user['karma_gain']"/> XP</span>
```
I think gaining negative karma it's not a "success". So instead should be yellow the badge. Red would be with badge-danger, as you prefer.

Comment (niyasraphy) [reply addons/website_profile/views/website_profile.xml:564]: thanks for the review and suggestion @MiquelRForgeFlow .
i am not sure whether i can make change such a change in stable.

@tde-banana-odoo is fine to apply this?
thanks

Comment (niyasraphy) [reply addons/website_profile/views/website_profile.xml:564]: @tde-banana-odoo if you have time, looking for your feedback on this

Comment (flch-odoo) [reply addons/website_profile/views/website_profile.xml:564]: You could use logic inside the badge (moving the `badge-success` to a `t-attf-class` attribute), which would likely be harmless in stable without requiring an additional t-if. 

```xml
<span class="badge badge-pill px-3 py-2" t-attf-class="'badge-success' if user['karma_gain']> 0 else ''">...</span>
```
I suppose something similar could be tried for the plus sign (in the `t-esc`).

Comment (niyasraphy) [reply addons/website_profile/views/website_profile.xml]: hi @flch-odoo ,
updated as per the suggestion.

positive gain:
Screenshot from 2023-02-14 20-53-23



negative gain:
Screenshot from 2023-02-14 20-53-47

will this be okay

**Review by MiquelRForgeFlow — APPROVED** [review]

File: addons/website_profile/views/website_profile.xml (hunk 1 above)
Comment (niyasraphy) [reply addons/website_profile/views/website_profile.xml]: ping @flch-odoo

**Comment by niyasraphy** [comment]
hi @flch-odoo ,
it it good to go now or any other comments.
thanks

**Comment by flch-odoo** [comment]
Thanks for the changes. I would also suggest to 
* keep the `badge-pill` in the "static" `class` attribute to make it clearer that it is not conditionally added
* use `<t t-out="('+' if user['karma_gain'] > 0 else '') + str(user['karma_gain'])"/>` instead of adding another `<t>` directive. (Note that I replaced `t-esc` with the preferred `t-out` (which also escapes content).

**Comment by niyasraphy** [comment]
thanks @flch-odoo ,
updated the pr as suggested.

**Comment by flch-odoo** [comment]
I'm sorry, @niyasraphy , I didn't mean to discard the part about `badge-success` vs `badge-danger`... You can keep those, but simplified, such as
```patch
- < ... t-attf-class="badge-pill #{ 'badge-success' if user['karma_gain'] > 0 else 'badge-warning'}">
+ < ... t-attf-class="'badge-success' if user['karma_gain'] > 0 else 'badge-warning'">
```

**Comment by niyasraphy** [comment]
hi @flch-odoo ,
ah i didn't read it correctly initially, that's why, sorry for it.

actually i moved badge-pill inside the t-attf-class because, 

keeping badge-pill outside

Screenshot from 2023-03-06 18-51-46


keeping badge-pill inside t-attf

Screenshot from 2023-03-06 18-50-55



thanks

**Comment by flch-odoo** [comment]
You are right, `class` is overridden by `t-attf-class` rendering. All of its content (`badge`, `px-3`...) should therefore be moved there to keep the styling.
Thanks!

**Comment by flch-odoo** [comment]
Looks good! 
I see the same problem in v14 though. Would you mind cherry-picking your fix on top of your 14.0 branch and re targeting the PR? 
Thanks for all your efforts!

**Comment by MiquelRForgeFlow** [comment]
@niyasraphy The PR you did in v14 (that was merged) didn't include the badge-success/badge-warning color fix. Will you do another PR or will you rebase this one or let it be? haha

**Comment by niyasraphy** [comment]
hi @MiquelRForgeFlow ,
i will close this PR and will create new pr for the master to apply the changes related to the color.
it seems changing the color in 14-16 will break the stable policy:  https://github.com/odoo/odoo/pull/114506

**Comment by MiquelRForgeFlow** [comment]
Well, maybe wait the merged commit reach master first haha

**Pull request closed without merging**